env.go

```diff
@@ -327,6 +327,12 @@ func Marshal(v interface{}) (EnvSet, error) {
 		}
 
 		for _, envKey := range envKeys {
+			if strings.Contains(envKey, "=") {
+				switch strings.ToLower(strings.SplitN(envKey, "=", 2)[0]) {
+				case "separator", "required", "default":
```

Comment:
Can you add a few test cases for `separator` and `required` as well?

Reply:
Test cases for separator and required were added in the last commit 👍